gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/EntityInteractionsFetcher.java

```diff
@@ -191,6 +281,72 @@ private EntityInteractionQueryResponse executeQueryRequest(
     return new EntityInteractionQueryResponse(entityInteractionQueryRequest, resultSet);
   }
 
+  private List<EntityInteractionQueryRequest> prepareInteractionIdsQueryRequest(
+      RequestContext requestContext,
+      EntitiesRequest request,
+      InteractionsRequest interactionsRequest,
+      boolean incoming) {
+    Set<String> otherEntityTypes = getOtherEntityTypes(interactionsRequest.getFilter());
+    if (otherEntityTypes.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    List<Filter> childFilters = new ArrayList<>();
+    childFilters.add(
+        QueryRequestUtil.createBetweenTimesFilter(
+            AttributeMetadataUtil.getTimestampAttributeId(metadataProvider, requestContext, SCOPE),
+            request.getStartTimeMillis(),
+            request.getEndTimeMillis()));
+
+    this.buildSpaceQueryFilterIfNeeded(requestContext, request.getSpaceId())
+        .ifPresent(childFilters::add);
+
+    List<String> idColumns =
+        getEntityIdColumnsFromInteraction(
+            DomainEntityType.valueOf(request.getEntityType()), !incoming);
+
+    // Group by the entity id column first, then the other end entity type for the interaction.
+    List<org.hypertrace.core.query.service.api.Expression> idExpressions =
+        idColumns.stream()
+            .map(QueryRequestUtil::createAttributeExpression)
+            .collect(Collectors.toUnmodifiableList());
+
+    // adding empty selections
```

Comment:
The selection isn't empty - it's a count. Why are we adding it? Initially I thought it was for passing validation, but I see we add more later. Comments are typically more helpful if they explain "why" rather than "what".

Also if leaving it, there's no reason to use a mutable list here since the following line is the only time we add to it.

Reply:
sure let's verify. That would be **very** surprising if it required an aggregation in a group by request.

Reply:
I was wrong this isn't needed. removing this